Add automated publishing of website with some extra cleanup

derberg opened the pull request:
First of all sorry for huge PR but it's because there was some necessary cleanup added. I will explain here what changes were done so it is easier to review:

- Rename of `hugo-fresh` theme to `hugo-asyncapi` and complete removal of readme and license from the old theme
- `resources` and `public` folders are removed from the repository and added to `.gitignore`. Public dir is completely regenerated on `hugo -t hugo-asyncapi`, also some files in `resources`. Correct me if I'm wrong but looks like `resources` is not needed at all, I don't see where generated scss files are used. I did not find any explanation here https://gohugo.io/getting-started/directory-structure/ so maybe it is just some legacy and we just leave it as it is, in `gitignore` and that is it?
- Add Netlify badge to landing page footer. Let me know if you like the location and the badge that I picked, others are listed here https://www.netlify.com/press/#badges
- Add netlify build status to Readme
- Some readme cleanup

---
Manual validation:

- Main website is built here on every change to `master` branch of my fork - https://keen-banach-449f3a.netlify.com/. You can see there is no Netlify footer (initial build was done `hugo -t hugo-fresh` as rename of the theme is not in the master)
- Link to preview you can see here https://github.com/derberg/website-2/pull/1 and you can see preview has a footer

---
Next steps:
Once this is merged I will do build setup on Netlify and configure preview generation

Related to #43

derberg committed:
Rename hugo-fresh to hugo-asyncapi and update Readme

derberg committed:
Remove public folder

derberg committed:
Remove resources that looks like they are not needed

derberg committed:
Add Netlify badge to the footer

derberg committed:
Add netlify build status badge to readme

derberg committed:
Add more updated to Readme

derberg committed:
Add dot to missing sentence to check previews work

`Subscribe to our newsletter to receive news about AsyncAPI` needed a dot + HTLM formatter

derberg commented:
Preview do not work here of course as the build command is changed

jonaslagoni reviewed (commented):
Honestly, I kinda like the light blue (green?) instead (https://www.netlify.com/img/global/badges/netlify-color-bg.svg)
![image](https://user-images.githubusercontent.com/13396189/73844781-8a526e00-4821-11ea-958c-6c9d0732f26f.png)
Not a big fan when you can only slightly see the box around the badge, but I guess this is just my preference :smile: 
![image](https://user-images.githubusercontent.com/13396189/73844894-c4237480-4821-11ea-981a-df8f373f6cfc.png)

I cannot review this further than saying it looks fine on the previews, have not used Hugo or Netlify before :smile:

fmvilas commented on public/_redirects:
```diff
@@ -1,3 +0,0 @@
-# These rules will change if you change your site’s custom domains or HTTPS settings
-
-https://asyncapi.com/* https://www.asyncapi.com/:splat 301!
```
Why are you removing this? This is to prevent the use of naked domains.

fmvilas reviewed (changes requested):
Left a comment. Also, SASS files are used [here](https://github.com/asyncapi/website/blob/master/themes/hugo-fresh/layouts/partials/css.html#L23). It took me a while to figure out too.

derberg committed:
Add _redirects file but in static dir and not public

derberg commented:
@fmvilas thanks for the info about sass, looks like it is part of the theme. So looks like we can ignore `resources` dir on local and have it in `.gitignore` as is regenerated during build time

@fmvilas regarding the `_redirects` I removed it as I removed whole `public` dir as it should not be in the repo. Sorry, didn't notice it is not regenarated in the dir during build. Now I recreated the file in `static` dir to have it copied over to `public` dir

@jonaslagoni this is why I provided link to other badges so others can share opinion as I don't have one :) I changed the logo, have a look

fmvilas reviewed (approved):
Awesome! :shipit:
